### DISCUSSION TITLE
Ability to follow links to symbols in Quick Documentation

### DISCUSSION THREAD

**Discussion by harry-xm** [redhat-developer/lsp4ij#496] [Q&A]
## Expected behavior

In GoLand and intellij-rust, links to other symbols can be clicked to jump to their documentation in place in the Quick Documentation popup.

![Screenshot 1](https://github.com/user-attachments/assets/76529dc6-1c04-487c-8984-9ddaaf5e72f4)

![Screenshot 2](https://github.com/user-attachments/assets/103b5cb4-4d78-433e-8c3f-950ea2061b3a)

## Actual behavior

With LSP4IJ and the respective official language servers, either the language server converts symbol references to URLs (e.g. rust-analyzer) and forces the user to leave the IDE for a browser, or the text in comments is displayed as is and is not clickable.

Besides, in a project that is private or not published to crates.io, generated links like https://docs.rs/pingora-lru/0.3.0/pingora_lru/struct.Lru.html would not be accessible.

![Screenshot 3](https://github.com/user-attachments/assets/3dcd3d09-965f-4828-bbfa-427ee64e89f0)

<img width="805" alt="Screenshot 4" src="https://github.com/user-attachments/assets/4fffbb73-b1c0-4763-879a-07976d6cc41e">

## Question

What can I do to implement symbol links in Quick Documentation properly and allow users to stay within the IDE?

**Comment by angelozerr** [comment]
If documentation  of rust-analyzer reports markdown url (see issue at https://github.com/redhat-developer/lsp4ij/issues/376) is it good for you?

**Reply by harry-xm** [reply]
You mean a file link? I'm expecting jump to documentation in place, not jump to source code.

**Comment by angelozerr** [comment]
Oh I see now what you want. Today click on http link open a Web Browser. 

You want to open the http link to a the Quick Documentation. I'm using a BrowerUtil from IJ,so perhaps it exists a settings in IJ to open the http page in the Quick documentation?

**Reply by harry-xm** [reply]
No, I don't really want an embedded browser. Maybe a screen recording will give you a better idea of what I mean:


https://github.com/user-attachments/assets/8ade6218-3b9b-48a5-9167-d3a25a051112


This combined with the Jump to Source button in the bottom right corner is the IntelliJ-native user experience that I would like to reproduce in LSP4IJ.

**Reply by angelozerr** [reply]
Ok thanks for your demo. I have no idea how to implement that. Please create an issue for that.
